lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -344,14 +353,20 @@ static bool decodeArrayElements(SILValue value,
   if (!elementType) return false;
 
   // Handle zero element pattern.
-  if (auto *tei = dyn_cast<TupleExtractInst>(value))
-    if (isZeroElementArrayAlloc(tei)) {
-      if (arrayInsts) {
-        arrayInsts->insert(tei);
-        arrayInsts->insert(tei->getOperand()->getDefiningInstruction());
+  if (auto *tei = dyn_cast<TupleExtractInst>(value)) {
```

Comment:
do we have a unit test on zero element pattern?

Reply:
Thanks, I will when I get a chance.